### PR TITLE
Apochromatic focusing example

### PR DESCRIPTION
**Pull request by cemitch99**
Added example input files for the drift-quad apochromatic focusing lattice shown in Fig. 4a of:

C. A. Lindstrom and E. Adli, Phys. Rev. Accel. Beams 19, 071002 (2016),
https://doi.org/10.1103/PhysRevAccelBeams.19.071002

- [x] Add draft of C++ input file and Python script
- [x] Update initial transverse beam moments based on email communication
- [x] Add analysis script (eg, check matching and emittance growth)
- [x] Add test to examples/CMakeLists
- [x] Add documentation

### CODE REVIEW

File: examples/apochromatic/input_apochromatic.in [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+beam.distribution = waterbag
+beam.sigmaX = 3.9984884770e-5
+beam.sigmaY = 3.9984884770e-5
+beam.sigmaT = 1.0e-3
+beam.sigmaPx = 2.6623538760e-5
+beam.sigmaPy = 2.6623538760e-5
+beam.sigmaPt = 2.0e-3
+beam.muxpx = 0.0
+beam.muypy = 0.0
+beam.mutpt = 0.1  #10% energy spread
```
Comment (ax3l) [inline examples/apochromatic/input_apochromatic.in]: Slightly easier to read for me :)
```suggestion
beam.kin_energy = 100.0e3  # 100 GeV nominal energy
beam.charge = 1.0e-9
beam.particle = electron
beam.distribution = waterbag
beam.sigmaX = 3.9984884770e-5
beam.sigmaY = 3.9984884770e-5
beam.sigmaT = 1.0e-3
beam.sigmaPx = 2.6623538760e-5
beam.sigmaPy = 2.6623538760e-5
beam.sigmaPt = 2.0e-3
beam.muxpx = 0.0
beam.muypy = 0.0
beam.mutpt = 0.1  # 10% energy spread
```

File: examples/apochromatic/run_apochromatic.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+distr = distribution.Waterbag(
+    sigmaX=3.9984884770e-5,
+    sigmaY=3.9984884770e-5,
+    sigmaT=1.0e-3,
+    sigmaPx=2.6623538760e-5,
+    sigmaPy=2.6623538760e-5,
+    sigmaPt=2.0e-3,
+    muxpx=0.0,
+    muypy=0.0,
+    mutpt=0.1,
```
Comment (ax3l) [inline examples/apochromatic/run_apochromatic.py]:
```suggestion
    sigmaX=3.9984884770e-5,
    sigmaY=3.9984884770e-5,
    sigmaT=1.0e-3,
    sigmaPx=2.6623538760e-5,
    sigmaPy=2.6623538760e-5,
    sigmaPt=2.0e-3,
    muxpx=0.0,
    muypy=0.0,
    mutpt=0.1,  # 10% energy spread
```

**Comment by cemitch99** [comment]
Emittance growth observed in ImpactX:  0.5%
Emittance growth quoted in Lindstrom et al:  0.96%

Note that Lindstrom (and ELEGANT) both use the non-canonical variables (x',y') while ImpactX always uses the canonical variables (px/p0,py/p0).  For small relative energy spread, these are essentially equivalent.  When energy spread becomes large, this difference becomes important.  This may be the source of the difference:  investigating this now.

I modified the analysis script to use (x',y') as momentum variables for computing the emittance, and this effect is far too small to account for this difference (~0.02%).

Solution:  The original ImpactX input used a 6D waterbag distribution, while Lindstrom et al assumes a Gaussian distribution in energy.  Changing to a 6D gaussian distribution and using the (x',y') variables above yields an emittance growth of 0.94%.

**Comment by ax3l** [comment]
Nice! Great reconstruction of events :detective: :smile:

File: examples/apochromatic/README.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
+.. _examples-apochromat:
+
+Apochromatic Drift-Quadrupole Beamline
+=======================================
```
Comment (ax3l) [inline examples/apochromatic/README.rst]:
```suggestion
Apochromatic Drift-Quadrupole Beamline
======================================
```

**Review by ax3l — APPROVED** [review]
Thank you, looks great!

I think we could add a plot script for this as well, we can do this as a follow-up PR.

**Pull request merged**